frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,154 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch, connect } from 'react-redux';
+import { match as RMatch } from 'react-router-dom';
+import { Table, TableHeader, TableBody, SortByDirection } from '@patternfly/react-table';
+import { FilterToolbar } from '@console/internal/components/filter-toolbar';
+import { getAlertsAndRules } from '@console/internal/components/monitoring/utils';
+import { monitoringSetRules } from '@console/internal/actions/ui';
+import { useURLPoll } from '@console/internal/components/utils/url-poll-hook';
+import { PrometheusRulesResponse, Rules } from '@console/internal/components/monitoring/types';
+import { PROMETHEUS_TENANCY_BASE_PATH } from '@console/internal/components/graphs';
+import { RootState } from '@console/internal/redux';
+import {
+  alertSeverityOrder,
+  alertStateOrder,
+  alertingRuleStateOrder,
+} from '@console/internal/reducers/monitoring';
+import { getFilteredRows } from '@console/internal/components/factory';
+import {
+  monitoringAlertRows,
+  monitoringAlertColumn,
+  alertFilters,
+} from './monitoring-alerts-utils';
+import './MonitoringAlerts.scss';
+
+type MonitoringAlertsProps = {
+  match: RMatch<{
+    ns?: string;
+  }>;
+};
+
+type StateProps = {
+  rules: Rules;
+  filters: { [key: string]: any };
+};
+
+type props = MonitoringAlertsProps & StateProps;
+
+const POLL_DELAY = 15 * 1000;
+const reduxID = 'devMonitoringAlerts';
+const textFilter = 'resource-list-text';
+
+export const MonitoringAlerts: React.FC<props> = ({ match, rules, filters }) => {
+  const [sortBy, setSortBy] = React.useState({});
+  const [rows, setRows] = React.useState([]);
+  const [collapsedRows, setCollapsedRows] = React.useState([]);
+  const dispatch = useDispatch();
+  const namespace = match.params.ns;
+  const [response, loadError, loading] = useURLPoll<PrometheusRulesResponse>(
+    `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`,
+    POLL_DELAY,
+    namespace,
+  );
+  const thanosRules = React.useMemo(
+    () => (!loading && !loadError ? getAlertsAndRules(response?.data).rules : []),
+    [response, loadError, loading],
+  );
+
+  React.useEffect(() => {
+    const sortThanosRules = _.sortBy(thanosRules, (rule) => alertingRuleStateOrder(rule));
+    dispatch(monitoringSetRules('devRules', sortThanosRules));
+  }, [dispatch, thanosRules]);
+
+  const filteredRules = React.useMemo(() => getFilteredRows(filters, alertFilters, rules), [
+    filters,
+    rules,
+  ]);
+
+  React.useEffect(() => {
+    const tableRows = monitoringAlertRows(filteredRules);
+    _.forEach(tableRows, (tr) => {
+      _.forEach(collapsedRows, (cr) => {
+        if (tr.cells[0].id === cr.cells[0].id) {
+          tr.isOpen = cr.isOpen;
+        }
+      });
+    });
+    setRows(tableRows);
+  }, [collapsedRows, filteredRules]);
+
+  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
```

Comment:
This would be simpler and more efficient if `collapsedRows` was an array of row keys instead of an array or row objects. Something like
```
  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
    if (isOpen) {
      setCollapsedRows([...collapsedRows, rowKey]);
    } else {
      _.pull(collapsedRows, rowKey);
      setCollapsedRows([...collapsedRows]);
    }
  };
```